staff-finding/scripts/component_filter.py

```
@@ -285,19 +327,80 @@ def filter_components(
+    merge_companion_coords: list[tuple[int, int]] = []
+    merge_companion_mask = np.zeros_like(labels, dtype=bool)
+
+    for s in survivors[1:]:
+        s_label, s_score, _s_sub, s_stats = s
+        if s_score < COMPANION_SCORE_FLOOR:
+            continue
+        comp_x = s_stats["x"]
+        comp_w = s_stats["w"]
+        comp_mask = labels == s_label
+        c_ys, c_xs = np.where(comp_mask)
+        comp_pixels = list(zip(c_xs.tolist(), c_ys.tolist()))
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)